src/Grpc.Net.Client/Internal/GrpcProtocolConstants.cs

```diff
@@ -103,15 +115,29 @@ static GrpcProtocolConstants()
             // but in case it isn't then don't include version in user-agent.
             if (assemblyVersion != null)
             {
-                userAgent += "/" + assemblyVersion.Version;
+                // 1.24.0
+                userAgent += $"/{assemblyVersion.Version} ";
```

Comment:
nit: I think assemblyVersion is actually a 4-part thing, so comment is wrong?
2.38.0.0?

note that Grpc.Core has a 3-part version number in user-agent string (since it doesn't extract the version number from the assemblyVersion but uses a different mechanism), but we can keep the 4-part format in grpc-dotnet for consistency with its older releases.

qq: it would be nice if the version number contained info about whether the version is a pre-release or a stable release (e.g. `2.38.0` vs `2.38.0-pre1`) since the distinction is quite important. 
Ideally we'd want to use this value as the version string (it's also what gets used as the nuget package version)
https://github.com/grpc/grpc-dotnet/blob/094532f0fbd3c9c877614beca7a73a797f3eba02/build/version.props#L5

Reply:
Good call! I've changed this to read from `AssemblyInformationalVersionAttribute`. I saw a comment around concerns with the git hash being in that string so I've stripped out the git hash from the generated user agent.